File: examples/genomedata_random_access.py
#!/usr/bin/env python
from __future__ import with_statement, division

"""Looks up track data in a Genomedata archive at positions on stdin

Usage: cat POSITIONS | ./genomedata_random_access.py GENOMEDATADIR TRACKNAME...

Expects lines on stdin of the form: chrom<whitespace>index

Prints the data value for the each trackname at each position in the format:
<value>TAB<value>TAB<value>
with a value for each trackname provided, in the order provided.
"""

import sys
import warnings
from genomedata import Genome

with Genome(sys.argv[1]) as genome:
    tracknames = sys.argv[2:]
    if not tracknames:
        print >>sys.stderr, "Using all tracks..."
        tracknames = genome.tracknames_continuous

    warnings.simplefilter("ignore")  # Ignore supercontig warnings
    for line in sys.stdin:
        chrom, index = line.strip().split()
        for trackname in tracknames:
            print str(genome[chrom][int(index), trackname]),

        print

File: examples/genomedata_random_access.pl
#!/usr/bin/env perl

# requires the Inline::Python module from CPAN
# and genomedata installed for Python
use Inline Python => <<'END';

from genomedata import Genome

def open_genomedata(filename):
    # simulate a context manager
    return Genome(filename).__enter__()

def close_genomedata(genome):
    return genome.__exit__(None, None, None)

def get_genomedata(genome, chrom, index, trackname):
    return float(genome[chrom][index, trackname])

END

my $genome = open_genomedata($ARGV[0]);
shift();

while (<STDIN>) {
    chomp();
    my ($chrom, $index) = split();
    foreach my $trackname (@ARGV) {
        print(get_genomedata($genome, $chrom, int($index), $trackname), " ");
    }
    print("\n");
}

close_genomedata($genome);
